VoxelSniperCore/src/main/java/com/github/kevindagame/VoxelBrushManager.java

```diff
@@ -125,58 +123,56 @@ public static VoxelBrushManager initialize() {
     public void registerSniperBrush(Class<? extends IBrush> clazz, String... handles) {
         Preconditions.checkNotNull(clazz, "Cannot register null as a class.");
         for (String handle : handles) {
-            brushes.put(clazz, handle.toLowerCase());
+            brushes.put(handle.toLowerCase(), clazz);
             brushHandles.add(handle.toLowerCase());
         }
     }
 
     /**
      * Retrieve Brush class via handle Lookup.
      *
-     * @param handle Case insensitive brush handle
+     * @param handle Case-insensitive brush handle
      * @return Brush class
      */
     public Class<? extends IBrush> getBrushForHandle(String handle) {
         Preconditions.checkNotNull(handle, "Brushhandle can not be null.");
-        if (!brushes.containsValue(handle.toLowerCase())) {
-            return null;
-        }
 
-        for (Map.Entry<Class<? extends IBrush>, String> entry : brushes.entries()) {
-            if (entry.getValue().equalsIgnoreCase(handle)) {
-                return entry.getKey();
-            }
-        }
-        return null;
+        return brushes.get(handle.toLowerCase());
     }
 
     /**
      * @return Amount of IBrush classes registered with the system under Sniper visibility.
      */
     public int registeredSniperBrushes() {
-        return brushes.keySet().size();
+        return (int) brushes.keySet().stream().distinct().count();
     }
 
     /**
      * @return Amount of handles registered with the system under Sniper visibility.
      */
     public int registeredSniperBrushHandles() {
-        return brushes.size();
+        return brushes.values().size();
     }
 
     /**
      * @param clazz Brush class
      * @return All Sniper registered handles for the brush.
      */
     public Set<String> getSniperBrushHandles(Class<? extends IBrush> clazz) {
-        return new HashSet<>(brushes.get(clazz));
+        Set<String> handles = new HashSet<>();
+        for (String key : brushes.keySet()) {
+            if (brushes.get(key).equals(clazz)) {
+                handles.add(key);
+            }
+        }
+        return handles;
     }
 
     /**
      * @return Immutable Multimap copy of all the registered brushes
      */
-    public Multimap<Class<? extends IBrush>, String> getRegisteredBrushesMultimap() {
-        return ImmutableMultimap.copyOf(brushes);
+    public Map<String, Class<? extends IBrush>> getRegisteredBrushesMap() {
+        return Map.copyOf(brushes);
```

Comment:
Does this need to be Immutable?

Reply:
I don't know, if this needs to be immutable, but it was immutable before.
But I think this makes sense, because is sespect, that the contents of the map can be changes from the outside otherwise.

Reply:
Yeah, I think it is fine this way.